arbor/fvm_lowered_cell.hpp

```diff
@@ -175,22 +179,40 @@ struct fvm_probe_data {
 // map to multiple probe representations within the cable_cell_group.
 
 struct probe_association_map {
-    // Keys are probe id.
+    // unique keys from multimap
+    std::vector<cell_address_type> keys(cell_member_predicate pred=all_probes) const {
+        std::vector<cell_address_type> res;
+        std::unordered_set<cell_address_type> seen;
+        for (const auto& [k, v]: data) {
+            if (!seen.count(k)) {
+                if (pred(k)) res.push_back(k);
+                seen.insert(k);
+            }
+        }
+        return res;
+    }
 
-    std::unordered_map<cell_member_type, probe_tag> tag;
-    std::unordered_multimap<cell_member_type, fvm_probe_data> data;
+    auto count(const cell_address_type& k) const { return data.count(k); }
 
-    std::size_t size() const {
-        arb_assert(tag.size()==data.size());
-        return data.size();
+    // Return range of fvm_probe_data values associated with probeset_id.
+    std::vector<const fvm_probe_data*> data_on(const cell_address_type& probeset_id) const {
```

Comment:
Maybe one can avoid creating and returning a vector here by just using a range constructed from the const_iterators form `equal_range`?

Reply:
1. I am a bit doubtful of the lifetime management; any modification to the underlying data structure might invalidate the range. Might just side effects from using too much Rust, though.
2. Can we even construct such a range? We get what is basically a range of `(k, v)` where `k` is constant...    We could use an iterator adapter for the occasion, but not a range, I think.

Reply:
You are right, if modification after usage of `data_on` is expected, then this would be an issue. I'm fine with leaving it as is, since this would likely be just a small optimization.